### PR TITLE
[xtensa] fixed few correctness issues in codegen + added new correctness tests

### PR DESCRIPTION
**Pull request by Aelphy**
also fixed few correctness issues, that were found using updated tests

### CODE REVIEW

**Comment by steven-johnson** [comment]
Where are the random inputs?

**Comment by Aelphy** [comment]
They are google related. Updating the description

**Comment by Aelphy** [comment]
A formatting check fails on the code that I didn't change, not sure what to do

**Comment by steven-johnson** [comment]
> They are google related. Updating the description

The description is still unchanged?

File: src/CodeGen_Xtensa_vectors.template.cpp [hunk 1]
```diff
@@ -1164,7 +1164,7 @@ template<>
 HALIDE_ALWAYS_INLINE void store_narrowing<native_vector_i16, int8_t, VECTOR_WIDTH_I16>(const native_vector_i16 &a, void *base, int32_t offset) {
     valign align = IVP_ZALIGN();
     xb_vecNx8 *__restrict ptr = (xb_vecNx8 *)((int8_t *)base + offset);
-    IVP_SANX8S_IP(a, align, ptr);
+    IVP_SANX8S_IP((a << 8) >> 8, align, ptr);
```
Comment (vksnk) [inline src/CodeGen_Xtensa_vectors.template.cpp:1167]: I am not sure I understand this part, could you please add comment?

Comment (Aelphy) [reply src/CodeGen_Xtensa_vectors.template.cpp:1167]: overflow in xtensa and x86 behaves differently. This move makes sure that we truncate and not saturate

**Review by vksnk — APPROVED** [review]

**Pull request merged**